src/main/java/uk/gov/hmcts/ccd/domain/model/definition/CaseType.java

```diff
@@ -142,6 +142,13 @@ public boolean hasEventId(String eventId) {
         return events.stream().anyMatch(event -> event.getId().equals(eventId));
     }
 
+    public CaseEvent findCaseEvent(String eventId) {
```

Comment:
Can this new method return optional just to avoid any NPEs? In EventTriggerService you can probably return null as all calls to EventTriggerService.findCaseEvent check for null.

Reply:
Good point. Done.